MLAgentsSDK/Assets/ML-Agents/Examples/GridWorld/Scripts/GridAgent.cs

```diff
@@ -9,14 +9,38 @@ public class GridAgent : Agent
     public float timeBetweenDecisionsAtInference;
     private float timeSinceDecision;
 
+    static int up = 1;
+    static int down = 2;
+    static int left = 3;
+    static int right = 4;
+
     public override void InitializeAgent()
     {
         academy = FindObjectOfType(typeof(GridAcademy)) as GridAcademy;
     }
 
     public override void CollectObservations()
     {
-
+        // Prevents the agent from picking an action that would make it collide with a wall
+        var positionX = (int) transform.position.x;
+        var positionZ = (int) transform.position.z;
+        var maxPosition = academy.gridSize - 1;
```

Comment:
can the grid size be changed at runtime? If not, move this to `InitializeAgent`

Reply:
It can be changed at runtime yes. It is one of the reset paramters.